### PR TITLE
feat: Add status codes [DEV-1582]

### PR DESCRIPTION
**Pull request by Toktar**

### CODE REVIEW

File: services/request_service.go [hunk 1]
```diff
@@ -30,137 +29,125 @@ func NewRequestService(didMethod string, ledgerService LedgerServiceI) RequestSe
...
-		return "", err
-	}
-
-	metadata, err := json.Marshal(&didResolution.Metadata)
-	if err != nil {
-		return "", err
+	resolutionMetadata, mErr1 := json.Marshal(didResolution.ResolutionMetadata)
+	didDoc, mErr2 := rs.didDocService.MarshallDID(didResolution.Did)
+	metadata, mErr3 := json.Marshal(&didResolution.Metadata)
+	if mErr1 != nil || mErr2 != nil || mErr3 != nil {
```
Comment (askolesov) [inline services/request_service.go:51]: It's not the best way of handling errors. I wouldn't chain them and would log the error.

File: services/request_service.go [hunk 2]
```diff
@@ -30,137 +29,125 @@ func NewRequestService(didMethod string, ledgerService LedgerServiceI) RequestSe
...
+		return createJsonResolutionInternalError(resolutionMetadata)
```
Comment (askolesov) [inline services/request_service.go:52]: Can we pass the actual error inside?

File: services/request_service.go [hunk 3]
```diff
@@ -30,137 +29,125 @@ func NewRequestService(didMethod string, ledgerService LedgerServiceI) RequestSe
...
-		return "", err
-	}
+	didDereferencing := rs.Dereference(did, dereferencingOptions)
 
-	dereferencingMetadata, err := json.Marshal(didDereferencing.DereferencingMetadata)
-	if err != nil {
-		return "", err
+	dereferencingMetadata, mErr1 := json.Marshal(didDereferencing.DereferencingMetadata)
+	metadata, mErr2 := json.Marshal(didDereferencing.Metadata)
+	if mErr1 != nil || mErr2 != nil {
```
Comment (askolesov) [inline services/request_service.go:73]: Why chain them?

Comment (askolesov) [reply services/request_service.go:73]: +1 for logging

**Review by askolesov — CHANGES_REQUESTED** [review]
Looks good overall, just a couple of remarks

**Comment by askolesov** [comment]
Oh, this is a draft. Just noticed

File: services/request_service.go (hunk 2 above)
Comment (Toktar) [reply services/request_service.go:53]: No, we can't. Only `internalError` message is available for Resolution Metadata

File: services/request_service.go (hunk 1 above)
Comment (Toktar) [reply services/request_service.go:51]: After a discussion, we decided to keep chain errors check, but add logging.
Done

**Comment by Toktar** [comment]
Added more logs. Discussed other comments with Alex

**Review by askolesov — APPROVED** [review]
Great

**Pull request merged**